collector/client.go

```diff
@@ -207,7 +215,19 @@ func (c *CacheObjectClient) GetInfos() (types.Counters, error) {
 }
 
 func (ch *connectionHandlerImpl) connect() (net.Conn, error) {
-	return net.Dial("tcp", fmt.Sprintf("%s:%d", ch.hostname, ch.port))
+	if ch.certificate != "" && ch.key != "" {
+		cert, err := tls.LoadX509KeyPair(ch.certificate, ch.key)
```

Comment:
Thanks for this PR, I think this is a good addition. however I'm not sure if squid server supports client certs for this work, does it?

If you meant to support when the cache manager port is behind a secure port as the `insecure` flag implies then I think the right approach might be what's explained here: https://gist.github.com/denji/12b3a568f092ab951456?permalink_comment_id=1695690#gistcomment-1695690

Please let me know if I misunderstood anything :)